runtime-sdk/src/dispatcher.rs

```diff
@@ -705,11 +764,20 @@ impl<R: Runtime + Send + Sync> transaction::dispatcher::Dispatcher for Dispatche
         method: &str,
         args: Vec<u8>,
     ) -> Result<Vec<u8>, RuntimeError> {
+        // Determine whether the method is allowed to access confidential state and provide an
+        // appropriately scoped instance of the key manager client.
+        let is_confidential_allowed = R::Modules::allowed_confidential_queries()
+            .iter()
+            .any(|&name| name == method);
```

Comment:
Right, so the reason I made this `Runtime::query_is_secure(method)` is because runtimes with a large number of queries called with about the same frequency could use a more efficient test like a `phf` lookup table, dynamic Huffman tree, or whatever. I originally had `Runtime::SECURE_QUERIES`, but it didn't seem incredibly easier for paratime devs one way or the other.

Also, I like this `R::Modules` approach. I didn't like how it was on `Runtime` for PT devs to configure, but I tried to rationalize your suggestion and ended up thinking it's a good thing that modules can't dictate what's secure. Maybe someone wants to run a c10l EVM that allows fetching encrypted storage (to "prove" that it's encrypted?) or disallowing calls because they don't like the KM implications. Perhaps an opt-out like `R::confidential_query_is_allowed` with a default implementation of `true` that gets ANDed with `is_confidential_allowed` suggested by the module would be useful.

Reply:
That's a good idea. Instead of placing it on the `Runtime` I would maybe add it to the core module's `Config` trait (currently I added a flag called `ALLOW_INTERACTIVE_READ_ONLY_TRANSACTIONS` that disables the whole thing -- defaults to disabled).

Reply:
Also about `phf`, we actually used this in the past but it was somewhat clunky. It was for method dispatch so maybe this needs a revisit now.